File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 9/8/2026, 12:36:41 am
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }

        unordered_map<char, int> sCount, tCount;

        for (int i = 0; i < s.length(); i++) {
            sCount[s[i]] += 1;
            tCount[t[i]] += 1;
        }

        return sCount == tCount;        
    }
};

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 9/8/2026, 12:36:46 am
class Solution(object):
    def isAnagram(self, s, t):
        if len(s)!=len(t):
            return False
        
        sd={}
        td={}

        for i in range(len(s)):
            sd[s[i]]=1+sd.get(s[i],0)
            td[t[i]]=1+td.get(t[i],0)

        return sd==td
        